src/api/api_parsers.cpp

```diff
@@ -244,22 +244,24 @@ extern "C" {
         }
         scoped_ptr<cmd_context>& ctx = mk_c(c)->cmd();
         std::string s(str);
-        std::istringstream is(s);
-        ctx->set_regular_stream(ous);
-        ctx->set_diagnostic_stream(ous);
+        ctx->reset_parser_streams(s);
+
+        ous = ctx->parser_ous();
```

Comment:
not sure why ownership of ous is organized this way now.

Reply:
The creation of the `stringstream` seemed to be expensive in my benchmarks, which motivated me to put it in the context to reuse it across calls. But beware that C++ is not my element. I could double check this if it feels amiss.

Reply:
I've moved the streams out of the context in 12c81d7. Whatever overhead it adds, I cannot distinguish it in my latest measures.